core/src/main/java/org/elasticsearch/index/query/PercolatorQuery.java

```diff
@@ -87,12 +86,23 @@ public void extractQueryTermsQuery(String extractedTermsFieldName, String unknow
             );
         }
 
+        /**
+         * @param percolateTypeQuery    A query that identifies all document containing percolator queries
+         */
+        public void setPercolateTypeQuery(Query percolateTypeQuery) {
+            this.percolateTypeQuery = Objects.requireNonNull(percolateTypeQuery);
+        }
+
         public PercolatorQuery build() {
+            // The query that selects which percolator queries will be evaluated by MemoryIndex:
```

Comment:
should we assert that either percolateTypeQuery or queriesMetaDataQuery is non null here?